src/workers/notifications/devCardUnlocked.ts

```diff
@@ -0,0 +1,37 @@
+import { messageToJson } from '../worker';
+import { NotificationBaseContext } from '../../notifications';
+import { NotificationType } from '../../notifications/common';
+import { NotificationWorker } from './worker';
+import { UserAction, UserActionType } from '../../entity';
+
+export const DEFAULT_DEV_CARD_UNLOCKED_THRESHOLD = 20;
+
+interface Data {
+  userId: string;
+}
+
+const worker: NotificationWorker = {
+  subscription: 'api.dev-card-unlocked-notification',
+  handler: async (message, con) => {
+    const data: Data = messageToJson(message);
+
+    const userAction = await con
+      .getRepository(UserAction)
+      .findOneBy({ userId: data.userId, type: UserActionType.DevCardUnlocked });
+
+    if (userAction) return null;
+
+    await con.getRepository(UserAction).insert({
+      userId: data.userId,
+      type: UserActionType.DevCardUnlocked,
+    });
```

Comment:
Why do we need the action? It's not really needed as you guarded yourself in the cdc condition and same notification for the same user can't be triggered more than once